### PR TITLE
Change Transcode API Task output format

### PR DESCRIPTION
**Pull request by leszko**
Related discussion in https://github.com/livepeer/catalyst/issues/397

### CODE REVIEW

File: pkg/data/task_result.go [hunk 1]
```diff
@@ -35,8 +35,18 @@ type TranscodeTaskOutput struct {
...
+
+type TranscodeFileTaskOutputPath struct {
+	Path string `json:"path,omitempty"`
+}
+
 type TranscodeFileTaskOutput struct {
-	VideoFilePath string `json:"videoFilePath"`
+	BaseUrl string                      `json:"baseUrl,omitempty"`
+	Hls     TranscodeFileTaskOutputPath `json:"hls,omitempty"`
+	Mp4     TranscodeFileTaskOutputMp4  `json:"mp4,omitempty"`
```
Comment (victorges) [inline pkg/data/task_result.go:49]: (yesterday I learned that) `omitempty` actually doesn't work if the field is a struct, so these need to be pointers instead if you want them actually omitted when not set
```suggestion
	Hls     *TranscodeFileTaskOutputPath `json:"hls,omitempty"`
	Mp4     *TranscodeFileTaskOutputMp4  `json:"mp4,omitempty"`
```

**Review by victorges — APPROVED** [review]

File: pkg/data/task_result.go [hunk 2]
```diff
@@ -35,8 +35,18 @@ type TranscodeTaskOutput struct {
 	Asset UploadTaskOutput `json:"asset,omitempty"`
 }
 
+type TranscodeFileTaskOutputMp4 struct {
+	Renditions []TranscodeFileTaskOutputPath `json:"renditions,omitempty"`
+}
```
Comment (victorges) [inline pkg/data/task_result.go:40]: Do we need this indirection? WDYT of just having the `Mp4` field as a `[]TranscodeFileTaskOutputPath`?

File: pkg/data/task_result.go (hunk 1 above)
Comment (leszko) [reply pkg/data/task_result.go:49]: Good point, thanks!

File: pkg/data/task_result.go (hunk 2 above)
Comment (leszko) [reply pkg/data/task_result.go:40]: I think it's fine. Then, the output will look like this (which I think is ok):
```json
"output": {
    "transcodeFile": {
        "url": "ipfs://bafybeifkjw5443v4wsp6q4cjuzboi7nt7i6n6iinyj7ii6fapqsuavg2hy",
        "hls": {
            "path": "video/index.m3u8"   
        },
        "mp4": [
             { "path": "video/static360p0.mp4" },
             { "path": "video/static720p0.mp4" },
             { "path": "video/static1440p0.mp4" }
         ]         
    }
}
```

**Pull request merged**